Model/SubscriptionManagement.php

```diff
@@ -0,0 +1,120 @@
+use Magento\Sales\Api\OrderManagementInterface;
+use Magento\Sales\Api\OrderRepositoryInterface;
+use Magento\Sales\Model\Order;
+use Paytrail\PaymentService\Api\SubscriptionLinkRepositoryInterface;
+use Paytrail\PaymentService\Api\SubscriptionRepositoryInterface;
+use Psr\Log\LoggerInterface;
+
+class SubscriptionManagement
+{
+    protected const STATUS_CLOSED = 'closed';
+    protected const ORDER_PENDING_STATUS = 'pending';
+
+    /**
+     * @var Session
+     */
+    protected $customerSession;
+
+    /**
+     * @var SubscriptionRepositoryInterface
+     */
+    protected $subscriptionRepository;
+
+    /**
+     * @var SubscriptionLinkRepositoryInterface
+     */
+    protected $subscriptionLinkRepository;
+
+    /**
+     * @var OrderRepositoryInterface
+     */
+    protected $orderRepository;
+
+    /**
+     * @var OrderManagementInterface
+     */
+    protected $orderManagementInterface;
+
+    /**
+     * @var RequestInterface
+     */
+    protected $request;
+
+    protected $customerRepository;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @param Session $customerSession
+     * @param SubscriptionRepositoryInterface $subscriptionRepository
+     * @param SubscriptionLinkRepositoryInterface $subscriptionLinkRepository
+     * @param OrderRepositoryInterface $orderRepository
+     * @param OrderManagementInterface $orderManagementInterface
+     * @param RequestInterface $request
+     * @param CustomerRepositoryInterface $customerRepository
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        Session $customerSession,
+        SubscriptionRepositoryInterface $subscriptionRepository,
+        SubscriptionLinkRepositoryInterface $subscriptionLinkRepository,
+        OrderRepositoryInterface $orderRepository,
+        OrderManagementInterface $orderManagementInterface,
+        RequestInterface $request,
+        CustomerRepositoryInterface $customerRepository,
+        LoggerInterface $logger
+    ) {
+        $this->customerSession = $customerSession;
+        $this->subscriptionRepository = $subscriptionRepository;
+        $this->subscriptionLinkRepository = $subscriptionLinkRepository;
+        $this->orderRepository = $orderRepository;
+        $this->orderManagementInterface = $orderManagementInterface;
+        $this->request = $request;
+        $this->customerRepository = $customerRepository;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param $subscriptionId
+     * @return string
+     */
+    public function cancelSubscription($subscriptionId)
+    {
+        try {
+            $subscription = $this->subscriptionRepository->get((int)$subscriptionId);
+            $orderIds = $this->subscriptionLinkRepository->getOrderIdsBySubscriptionId((int)$subscriptionId);
+            $params = $this->request->getParams();
+
+            $customer = $this->customerRepository->get($params['username']);
```

Comment:
You can get the customer via first checking if the customer is logged in: $this->customerSession->isLoggedIn() and then  $customer = $this->customerSession->getCustomer();

Reply:
Done